src/main/java/cofh/thermaldynamics/util/RecipeCover.java

```diff
@@ -11,13 +11,14 @@
 import net.minecraftforge.common.ForgeHooks;
 
 import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
 
 public class RecipeCover implements IRecipe {
 
 	public static final RecipeCover instance = new RecipeCover();
 
 	@Override
-	public boolean matches(@Nonnull InventoryCrafting craft, @Nonnull World world) {
```

Comment:
Thermal expansions SawMill calls `recipe.matches()` with a null-world  on startup and some other situations. This would cause a crash in development environments and might lead to the expection that world is never null - which is not the case.